encryptor/searchable_query_filter.go

```diff
@@ -123,39 +128,31 @@ func (filter *SearchableQueryFilter) filterTableExpressions(statement sqlparser.
 	}
 }
 
-func (filter *SearchableQueryFilter) filterComparisonExprs(statement sqlparser.Statement) []*sqlparser.ComparisonExpr {
+func (filter *SearchableQueryFilter) filterComparisonExprs(statement sqlparser.Statement, defaultTable *AliasedTableName, aliasedTables AliasToTableMap) []SearchableExprItem {
 	// Walk through WHERE clauses of a SELECT statements...
 	whereExprs, err := getWhereStatements(statement)
 	if err != nil {
 		logrus.WithError(err).Debugln("Failed to extract WHERE clauses")
 		return nil
 	}
-	// ...and find all eligible comparison expressions in them.
+
 	var exprs []*sqlparser.ComparisonExpr
 	for _, whereExpr := range whereExprs {
-		comparisonExprs, err := getEqualComparisonExprs(whereExpr)
+		comparisonExprs, err := filter.getEqualComparisonExprs(whereExpr, defaultTable, aliasedTables)
 		if err != nil {
 			logrus.WithError(err).Debugln("Failed to extract comparison expressions")
 			return nil
 		}
+
 		exprs = append(exprs, comparisonExprs...)
 	}
-	return exprs
-}
 
-func (filter *SearchableQueryFilter) filterComparisons(exprs []*sqlparser.ComparisonExpr, defaultTable *AliasedTableName, aliasedTables AliasToTableMap) []SearchableExprItem {
-	filtered := make([]SearchableExprItem, 0, len(exprs))
```

Comment:
why you decided to remove pre-allocation? with allocation empty slice golang will create a new buffer with 2 elements and after that increase further by power of 2. 2...4...8...16. So, we can meet 1-3 new allocation with copying current elements to new one, which looks inefficient.

Reply:
You definitely right, just change the function semantics and forgot to init slice in proper way.